### DISCUSSION TITLE
Using Bluetooth

### DISCUSSION THREAD

**Discussion by Jakethe1stdog** [DesktopECHO/Pi-hole-for-Android#92] [Q&A] [answered]
I've successfully installed pi deploy and it works great, but I'd like to use the Bluetooth functionality of my phone. When listing the Bluetooth adapter using the bluetoothctl command no adapters can be found, but when listing the adapters via rfkill it can see the bluetooth adapter of my phone. Is it possible to enable or mount the Bluetooth adapter somehow?

**Comment by DesktopECHO** [comment]
Hi... If you enable bluetooth tethering Android will assign an IP to the BT interface.

**Reply by Jakethe1stdog** [reply]
Sorry I was a bit unclear.

I want my device's Bluetooth controller to show up when I search for it in the terminal. Right now I get the message: "No default controller available" when looking for it. Is this possible?

**Reply by DesktopECHO** [reply]
When running `ip a` (for example) to see your network interfaces, the Bluetooth interface doesn't show up because Android has not given it an IP address.

You won't see the Bluetooth TCP/IP interface until you either enable tethering or Bluetooth PAN... see here for an example:
https://android.stackexchange.com/questions/12572/are-there-any-android-phones-that-allow-bluetooth-pan-off-the-shelf

**Reply by Jakethe1stdog** [reply]
I've enabled it in the android settings but I still don't see it listed.

**Reply by Jakethe1stdog** [reply]
I'm trying to setup the Bluetooth integration for Home Assistant which installed through Pi Deploy. The setup wizard doesn't work because it doesn't detect any Bluetooth adapters. I'm running this through an S7 edge which has a bcm4359 WIFI/Bluetooth combo chip. I've ran hciconfig, bluetoothctl, dmesg grep -i blue, and rfkill with Bluetooth tethering on and off which all return nothing except for rfkill, which does show a Bluetooth device. I've also tried connecting a Bluetooth adapter via an OTG cable with no output as well, but it does show up in lsusb. I've also tried debian 10 and 12 which also doesn't Work.

I'd really appreciate some help. Thank you.

**Reply by DesktopECHO** [reply, marked as answer]
Unfortunantely I don't think you're going to be able to achieve what you want to accomplish, not easily at least.  

Android uses a different bluetooth stack ([Fluoride](https://android.googlesource.com/platform/system/bt/+/289a49814aef7f0f0bb98aac8246080abdfeac01/README.md) or [Gabeldorsche](https://www.xda-developers.com/android-13-gabeldorsche-bluetooth-stack/)) than what's available in normal Linux.  That's why the container isn't responding to BT commands in the way you'd expect.

You can use the `unchroot` command to escape the container and play with the Android BT stack directly, for example try `sudo unchroot dumpsys bluetooth_manager`

**Reply by Jakethe1stdog** [reply]
Ah I see, that's unfortunate.

Thank you for the reply.
